vector/src/main/java/im/vector/app/features/home/room/detail/composer/MessageComposerViewModel.kt

```diff
@@ -472,14 +472,16 @@ class MessageComposerViewModel @AssistedInject constructor(
                             Unit
                         }
                         is ParsedCommand.UpgradeRoom                       -> {
-                            _viewEvents.post(
-                                    MessageComposerViewEvents.ShowRoomUpgradeDialog(
-                                            parsedCommand.newVersion,
-                                            room.roomSummary()?.isPublic ?: false
-                                    )
-                            )
-                            _viewEvents.post(MessageComposerViewEvents.SlashCommandResultOk(parsedCommand))
-                            popDraft()
+                            viewModelScope.launch {
+                                _viewEvents.set(
+                                        MessageComposerViewEvents.ShowRoomUpgradeDialog(
+                                                parsedCommand.newVersion,
+                                                room.roomSummary()?.isPublic ?: false
+                                        )
+                                )
+                                _viewEvents.set(MessageComposerViewEvents.SlashCommandResultOk(parsedCommand))
```

Comment:
The pb was we were posting 2 ViewEvents in a row?
Maybe keep only the first one and assume the receiver will do the same treatment than when the second one is sent? This will avoid us to add new API to the DataSource.

Reply:
I tested this but the receiver doesn't accomplish the same. Without this ok command, future messages won't send. Even if we handle things on the receiver side, it's only confusing the intent on both sides.

I do also think that having access to a non async `emit` in the data source is very important, otherwise we'd always have to find workarounds like this whenever we want to post two events in succession

Reply:
Technically, we should't be able to have multiple Events sent in a row before the previous one is processed, otherwise this should more be part of ViewState. 
That being said, I'm not against changing the API to only use `emit` instead of `tryEmit`